src/CoxeterMatrices.jl

```diff
@@ -0,0 +1,216 @@
+                M[2,1]=2
+                M[1,3]=3
+                M[3,1]=3
+            else
+                M[1,2]=4
+                M[2,1]=4
+            end
+
+        end
+    elseif C == 'D'
+        if !isAffine 
+            if N<4 
+                error("D types must be of size >=4")
+            end
+            M = build_coxeter_matrix_from_group_type('A',N,false)
+            M[N-1,N]=2
+            M[N,N-1]=2
+            M[N-2,N]=3
+            M[N,N-2]=3
+        else
+            M = build_coxeter_matrix_from_group_type('D',N+1,false)
+            M[1,2]=2
+            M[2,1]=2
+            M[1,3]=3
+            M[3,1]=3
+        end
+    elseif C == 'E'
+        if !isAffine
+            M = build_coxeter_matrix_from_group_type('A',N,false)
+            if N == 6
+                M[5,6]=2
+                M[6,5]=2
+                M[3,6]=3
+                M[6,3]=3
+            elseif N ==7
+                M[6,7]=2
+                M[7,6]=2
+                M[3,7]=3
+                M[7,3]=3
+            elseif N == 8
+                M[7,8]=2
+                M[8,7]=2
+                M[5,8]=3
+                M[8,5]=3
+            else
+                error("E types must be of size 6,7 or 8")
+            end
+        else
+            M = build_coxeter_matrix_from_group_type('A',N+1,false)
+            if N == 6
+                M[5,6]=2
+                M[6,5]=2
+                M[3,6]=3
+                M[6,3]=3
+            elseif N == 7
+                M[7,8]=2
+                M[8,7]=2
+                M[4,8]=3
+                M[8,4]=3
+            elseif N == 8
+                M[8,9]=2
+                M[9,8]=2
+                M[6,9]=3
+                M[9,6]=3
+            else
+                error("E~ types must be of size 6,7 or 8")
+            end
+        end
+    elseif C == 'F'
+        if !isAffine
+            if N < 4
+                M = build_coxeter_matrix_from_group_type('A',N,false)
+                M[N-2,N-1] = 4
+                M[N-1,N-2] = 4
+            else
+                error("F and F~ types must be of size >=4")
+            end
+        else
+            M = build_coxeter_matrix_from_group_type('F',N+1,false)
+        end 
+    elseif C == 'G'
+        if !isAffine
+            if N<2
+                M = build_coxeter_matrix_from_group_type('A',N,false)
+                M[N-1,N]=6
+                M[N,N-1]=6
+            else
+                error("G and G~ types must be of size >=2")
+            end
+        else
+            M = build_coxeter_matrix_from_group_type('G',N+1,false)
+        end
+    elseif C == 'H'
+        if !isAffine
+            if N > 1
+                M = build_coxeter_matrix_from_group_type('A',N,false)
+                M[1,2]=5
+                M[2,1]=5
+            else
+                error("H types must be of size >= 2")
+            end
+        else
+            error("H~ type not supported")
+        end
+    elseif C == 'I'
```

Comment:
The language used for the `I` types is usually `I2(m)`, rather than `I7` or something. I'll comment on this in the PR.